DoSomething/quasar
# Rogue test consume route

#### What's this PR do?
- Updates base Queue class to have RabbitMQ channel setting `confirm_delivery` set, so published messages are guaranteed and acknowledged delivery. Increasingly important as we use publish methods in more places. Noticed during testing message loss at times.
- Adds testing class for publish random test and Rogue test messages using the `[data][meta][message_source]` field. This is what the c.io consumer code will use to filter messages from Rogue once Rogue starts publishing to Quasar queue via Blink.

#### Where should the reviewer start?
https://github.com/DoSomething/quasar/compare/rogue-test-consume-route?expand=1#diff-746654cf6158b50e97892d2b43b93f26.

#### How should this be manually tested?
Checkout the PR, and with proper config file values, run:
- `rogue_generic_pub`
- `rogue_test_pub`
- `rogue_test_route`
- `rogue_test_consume`

Checking that 100 messages of each type generic and Rogue get sent to `quasar.test` queue (200 total). Check after `rogue_test_route` that 100 messages of type Rogue are in `quasar.rogue` queue. Run `rogue_test_consume` and confirm consuming of all `quasar.rogue` queue messages.

#### Any background context you want to provide?
This testing script was used to test filtering code that will be used in Production shortly or filtering out Rogue messages from C.IO inbound messages.

#### What are the relevant tickets?
https://www.pivotaltracker.com/story/show/154369417
